src/components/OrderDetailWidget/OrderDetailWidget.tsx

```diff
@@ -1,73 +1,169 @@
-import React, { FC } from "react";
+import React, { FC, useMemo, useContext } from "react";
 import { useTranslation } from "react-i18next";
 import { useHistory } from "react-router-dom";
 
+import { FullOrder } from "@airswap/typescript";
 import { Web3Provider } from "@ethersproject/providers";
 import { useToggle } from "@react-hookz/web";
 import { useWeb3React } from "@web3-react/core";
+import { UnsupportedChainIdError } from "@web3-react/core";
 
-import { addDays } from "date-fns";
+import { BigNumber } from "bignumber.js";
+import compareAsc from "date-fns/compareAsc";
+import { ethers } from "ethers";
 
 import { nativeCurrencyAddress } from "../../constants/nativeCurrency";
+import { InterfaceContext } from "../../contexts/interface/Interface";
+import stringToSignificantDecimals from "../../helpers/stringToSignificantDecimals";
+import switchToEthereumChain from "../../helpers/switchToEthereumChain";
+import useInsufficientBalance from "../../hooks/useInsufficientBalance";
+import useTokenInfo from "../../hooks/useTokenInfo";
 import { OrderStatus } from "../../types/orderStatus";
 import { OrderType } from "../../types/orderTypes";
 import FeeModal from "../InformationModals/subcomponents/FeeModal/FeeModal";
+import { ButtonActions } from "../MakeWidget/subcomponents/ActionButtons/ActionButtons";
 import Overlay from "../Overlay/Overlay";
 import SwapInputs from "../SwapInputs/SwapInputs";
 import { Container, StyledInfoButtons } from "./OrderDetailWidget.styles";
 import ActionButtons from "./subcomponents/ActionButtons/ActionButtons";
 import OrderDetailWidgetHeader from "./subcomponents/OrderDetailWidgetHeader/OrderDetailWidgetHeader";
 
-// Dummy data, this data will come from store later.
-const expiry = addDays(new Date(), 1);
-const orderStatus = OrderStatus.open;
-const orderType = OrderType.private;
-const recipientAddress = nativeCurrencyAddress;
-const transactionLink =
-  "https://etherscan.io/tx/0x53ddb90435ab46d96035da71b4ff8a26bb0a682a7f2327aaeb8ff3848406204f";
+interface OrderDetailWidgetProps {
+  status: "idle" | "not-found" | "open" | "taken" | "canceled";
+  order: FullOrder;
+}
 
-const OrderDetailWidget: FC = () => {
+const OrderDetailWidget: FC<OrderDetailWidgetProps> = ({ status, order }) => {
   const { t } = useTranslation();
   const history = useHistory();
-  const { account } = useWeb3React<Web3Provider>();
-
+  const { setShowWalletList } = useContext(InterfaceContext);
+  const { active, account, error: web3Error } = useWeb3React<Web3Provider>();
   const [showFeeInfo, toggleShowFeeInfo] = useToggle(false);
+  const senderToken = useTokenInfo(order.senderToken);
+  const signerToken = useTokenInfo(order.signerToken);
+  const hasInsufficientTokenBalance = useInsufficientBalance(
+    signerToken,
+    ethers.utils.formatUnits(order.signerAmount, signerToken?.decimals!)
```

Comment:
`signerToken?.decimals!` what if signerToken is null? Things will break.

Reply:
checked out your implementation in MakeWidget. do you see the optimal solution as something like ```ethers.utils.formatUnits(order.signerAmount, signerToken?.decimals || "ether")``` or would it be better to handle in useTokenInfo, by always sending a ```defaultAddress```?

Reply:
`decimals` should be a number, we could default it to 18 but then the user could potentially see a jumping number on the first frame.

And when `useTokenInfo` defaults to a `defaultAddress` you'll have the same problem. A token that's initially ether and then jumps to some other token. So the best way is to default to `undefined`:

```
const signerAmount = useMemo(() => {
    return signerToken ? ethers.utils.formatUnits(order.signerAmount, signerToken.decimals) : undefined
  }, [order, signerToken]);
  const hasInsufficientTokenBalance = useInsufficientBalance(
    signerToken,
    signerAmount,
  );
```

And then adjust hasInsufficientTokenBalance line 12: `amount?: string`.

And then to make things more clean I would add a `useFormattedTokenAmount` hook:

```
const useFormattedTokenAmount = (
  amount?: string,
  decimals?: number,
): string | undefined => {
  return useMemo(() => {
    if (!decimals || !amount) {
      return undefined;
    }

    return ethers.utils.formatUnits(amount, decimals);
  }, [amount, decimals]);
};

export default useFormattedTokenAmount;
```

And then OrderDetailWidget.tsx will look a lot cleaner.

```
  const signerAmount = useFormattedTokenAmount(order.signerAmount, signerToken?.decimals);
  const hasInsufficientTokenBalance = useInsufficientBalance(
    signerToken,
    signerAmount,
  );
```

You can do the same for `senderAmount` and then reduce some code for `const baseAmount` and `const quoteAmount`.